awx/api/permissions.py

```diff
@@ -250,3 +251,16 @@ def has_permission(self, request, view):
 class WebhookKeyPermission(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         return request.user.can_access(view.model, 'admin', obj, request.data)
+
+
+class AnalyticsPermission(permissions.BasePermission):
+    """
+    Allows GET/POST/OPTIONS to system admins and system auditors.
+    """
+
+    def has_permission(self, request, view):
+        if not (request.user and request.user.is_authenticated):
+            return False
+        if request.method in ["GET", "POST", "OPTIONS"]:
```

Comment:
In the view you already check for GET/POST/OPTIONS and 500 if its not one of them. Thinking about flexibility in the future, should we change this and the two likes below it so that we return False if the user is not authed and then just make the method:
```
return request.user.is_superuser or request.user.is_system_auditor
```
Or do you envision one day having a delete function which could be called by a superuser but not callable by an auditor?

Reply:
@john-westcott-iv yes, that's possible, also there will be PUT/PATCH in 2.5 (I suppose, the cloud API is quite non-standard) - Automation Calculator page could send data to cloud, and that's for admins only